contributors/design-proposals/raw-block-pv.md

```diff
@@ -0,0 +1,559 @@
+
+WORKFLOW:
+
+ADMIN:
+
+```
+kind: PersistentVolume
+apiVersion: v1
+metadata:
+  name: local-raw-pv
+spec:
+  volumeType: block
+  capacity:
+    storage: 100Gi
+  local:
+    path: /dev/xvdc
+  accessModes:
+    - ReadWriteOnce
+  persistentVolumeReclaimPolicy: Delete
+```
+
+## UC2: 
+
+DESCRIPTION: 
+
+A user uses a container on virtual machine on hypervisor such as KVM, VMware and wishes to use raw block device for databases such as MariaDB. 
+
+BACKGROUND:
+
+Using hypervisor's device passthrough feature(KVM: device passthrough, VMware: RawDeviceMapping), storage admin can expose raw block device into virtual machine then user can consume it via PV and PVC. Also dynamic provisioning could work if external-provisioner support this.
+
+WORKFLOW:
+
+ADMIN:
+
+* Admin creates a disk and attach it to server and pass-through the disk to KVM guest which Kubelet node1 is working inside.
+* Admin creates a PV using the path to specify the device location.
+
+```
+kind: PersistentVolume
+apiVersion: v1
+metadata:
+  name: raw-pv
+spec:
+  volumeType: block
+  capacity:
+    storage: 100Gi
+  accessModes:
+    - ReadWriteOnce
+  persistentVolumeReclaimPolicy: Delete
+  fc:
+    targetWWNs: ['500a0982991b8dc5']
+    lun: 2
+```
+
+* Admin adds "raw-disk" label to the kubelet node1
+
+```
+% kubectl label nodes <node1-ip> type=raw-disk
+```
+
+USER:
+
+* User creates a persistent volume claim with volumeType: block option.
+
+```
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: raw-pvc
+spec:
+  volumeType: block
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 80Gi
+```
+
+*   User creates a Pod yaml which uses raw-pvc PVC and selects a node that Admin attached raw disk using nodeSelector option.
+
+```
+apiVersion: v1
+kind: Pod
+metadata:
+  name: my-db
+spec:
+    containers
+    - name: mysql
+      image: mysql
+      volumeMounts:
+      - name: my-db-data
+	mountPath: /dev/xvda
+    nodeSelector:
+      type: raw-disk
+    volumes:
+    - name: my-db-data
+      persistentVolumeClaim:
+	claimName: raw-pvc
+```
+
+## UC3: 
+
+DESCRIPTION: 
+
+A developer wishes to enable their application to use a local raw block device as the volume for the container. The admin has already created PVs that the user will bind to by specifying 'block' as the volume type of their PVC.
+
+BACKGROUND:
+
+For example, an admin has already created the devices locally and wishes to expose them to the user in a consistent manner through the 
+Persistent Volume API.
+
+WORKFLOW:
+
+USER:
+
+```
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: local-raw-pvc
+spec:
+  volumeType: block
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 80Gi
+```
+
+```
+apiVersion: v1
+kind: Pod
+metadata:
+  name: my-db
+spec:
+    containers:
+    - name: mysql
+      image: mysql
+      volumeMounts:
+      - name: my-db-data
+	mountPath: /var/lib/mysql/data
```

Comment:
PV defines *source* of the volume. How it is exposed to the pod? Above you wrote that it's not `mountPath`. So how is a MySQL in a container supposed to find the device?